app/src/main/java/de/westnordost/streetcomplete/quests/smoothness/AddSmoothnessForm.kt

```diff
@@ -0,0 +1,166 @@
+            }
+        }
+        Smoothness.VERY_BAD -> when (highwayTag) {
+            in ALL_PATHS_EXCEPT_STEPS -> R.string.quest_smoothness_description_very_bad_path
+            else -> R.string.quest_smoothness_description_very_bad_road
+        }
+        Smoothness.HORRIBLE -> when (highwayTag) {
+            in ALL_PATHS_EXCEPT_STEPS -> R.string.quest_smoothness_description_horrible
+            else -> R.string.quest_smoothness_description_horrible
+        }
+        Smoothness.VERY_HORRIBLE -> when (highwayTag) {
+            in ALL_PATHS_EXCEPT_STEPS -> R.string.quest_smoothness_description_very_horrible
+            else -> R.string.quest_smoothness_description_very_horrible
+        }
+        Smoothness.IMPASSABLE -> when (highwayTag) {
+            in ALL_PATHS_EXCEPT_STEPS -> R.string.quest_smoothness_description_impassable
+            else -> R.string.quest_smoothness_description_impassable
+        }
+    }
+
+    private fun getImage(smoothness: Smoothness) = when (smoothness) {
+        Smoothness.EXCELLENT -> when (surfaceTag) {
+            "paving_stones" -> R.drawable.surface_paving_stones_excellent
+            else -> R.drawable.surface_asphalt_excellent
+        }
+        Smoothness.GOOD -> when (surfaceTag) {
+            "paving_stones" -> R.drawable.surface_paving_stones_good
+            else -> R.drawable.surface_asphalt_good
+        }
+        Smoothness.INTERMEDIATE -> when (surfaceTag) {
+            "paving_stones" -> R.drawable.surface_paving_stones_intermediate
+            "sett" -> R.drawable.surface_sett_intermediate
+            "compacted" -> R.drawable.surface_compacted_intermediate
+            else -> R.drawable.surface_asphalt_intermediate
+        }
+        Smoothness.BAD -> when (surfaceTag) {
+            "paving_stones" -> R.drawable.surface_paving_stones_bad
+            "sett" -> R.drawable.surface_sett_bad
+            "gravel" -> R.drawable.surface_gravel_bad
+            "compacted" -> R.drawable.surface_compacted_bad
+            else -> R.drawable.surface_asphalt_bad
+        }
+        Smoothness.VERY_BAD -> when (surfaceTag) {
+            "paving_stones" -> R.drawable.surface_paving_stones_very_bad
+            "sett" -> R.drawable.surface_sett_very_bad
+            else -> R.drawable.surface_asphalt_very_bad
+        }
+        // TODO: need images for the values below (and replace the ones above)
+        Smoothness.HORRIBLE  -> R.drawable.surface_paved_area
+        Smoothness.VERY_HORRIBLE -> R.drawable.surface_paved_area
+        Smoothness.IMPASSABLE -> R.drawable.surface_paved_area
+    }
+
+    private fun surfaceWrong() {
+        val surfaceType = Surface.values().find { it.osmValue == surfaceTag }
+
+        if (surfaceType != null) {
+            showWrongSurfaceDialog(surfaceType)
+        } else {
+            // this should not happen: smoothness quest is not asked for unknown surfaces
+            composeNote()
+        }
```

Comment:
```suggestion
        val surfaceType = Surface.values().find { it.osmValue == surfaceTag }!!
        showWrongSurfaceDialog(surfaceType)
```

If this should not happen, it is a programming error and thus the app should crash so that this error can be reported.